src/app/shared/components/template/components/task-progress-bar/task-progress-bar.component.ts

```diff
@@ -1,81 +1,175 @@
-import { Component, EventEmitter, Input, OnInit, Output } from "@angular/core";
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from "@angular/core";
 import { TaskService } from "src/app/shared/services/task/task.service";
-import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+import {
+  getBooleanParamFromTemplateRow,
+  getStringParamFromTemplateRow,
+} from "src/app/shared/utils";
 import { TemplateBaseComponent } from "../base";
 import { TemplateFieldService } from "../../services/template-field.service";
 import { AppDataService } from "src/app/shared/services/data/app-data.service";
 import { IProgressStatus } from "src/app/shared/services/task/task.service";
 import { CampaignService } from "src/app/feature/campaign/campaign.service";
+import { Subscription, debounceTime } from "rxjs";
+import { DynamicDataService } from "src/app/shared/services/dynamic-data/dynamic-data.service";
+
+interface ITaskProgressBarParams {
+  /**
+   * TEMPLATE PARAMETER: task_group_data.
+   * The name of the task group to track the progress of (through the completion of its subtasks)
+   */
+  taskGroupDataList: string;
```

Comment:
nit(non-blocking) - as mentioned previously ,can probably omit the word `task` and would also be good to specify it's the data list `name` (easily confused with entire list). So would suggest `dataListName` or `groupDataListName` (not sure how much group is implied or not)